Tests/AlicerceTests/Network/NetworkStackTestCase.swift

```diff
@@ -0,0 +1,145 @@
+import XCTest
+@testable import Alicerce
+
+class NetworkStackTestCase: XCTestCase {
+
+    private typealias Resource = UUID
+    private typealias Payload = String
+    private typealias Response = Int
+
+    private enum MockNetworkError: Error { case 🔥 }
+    private enum MockDecodeError: Error { case 💣, 🧨 }
+
+    private typealias NetworkStack = MockNetworkStack<Resource, Payload, Response, MockNetworkError>
+    private typealias NetworkDecoding<T> = ModelDecoding<T, Payload, Response>
+
+    private var stack: NetworkStack!
+
+    private var resource: Resource!
+    private let networkValue = Network.Value(value: "🌍", response: 1337)
+
+    override func setUpWithError() throws {
```

Comment:
Shouldn't we've a call to super? Same with `tearDownWithError`

Reply:
For some reason the new autogenerated `XCTestCase` Unit Test templates don't have default `super` calls for `setUpWithError()` and `tearDownWithError()`, and as such I didn't implement them either 😅.

My guess is that it should no longer be necessary, because the "old" `setUp` and `tearDown` are still called under the hood, according to the documentation:

> `setUpWithError()`
>
> Before each test begins, XCTest calls setUpWithError(), followed by setUp(). If state preparation might throw errors, override setUpWithError(). XCTest marks the test failed when it catches errors, or skipped when it catches XCTSkip.

> `tearDownWithError()`
>
> After each test completes, XCTest calls tearDown(), followed by tearDownWithError(). If state cleanup might throw errors, override tearDownWithError(). XCTest marks the test failed when when it catches errors, or skipped when it catches XCTSkip.

So in essence I think Apple is telling us that we should use this to setup/tear down our tests and take advantage of the new throwing capabilities to mark tests as failed if necessary. Furthermore, I get the impression that the default (`super`) implementations for these two new methods don't do anything (otherwise Apple would add the `super` calls).

However, if you prefer I can add them as I think it won't hurt. It will add extra verbosity though. 😄

Reply:
Well, we don't have an hierarchy of more than one level (we're always subclassing `XCTestCase` directly) so it should be fine to not add them.